### DISCUSSION TITLE
Windows version not working - no error message

### DISCUSSION THREAD

**Discussion by vtontodonato** [juanmc2005/diart#193] [Q&A]
I cannot manage to let it works on windows ... after all settings; the call to
 **diart.stream "./audio/test.wav" --output "./test_res.txt"** does not create any file. What is wrong? The prompt output is

 C:\Users\valen\.conda\envs\diart\lib\site-packages\pyannote\audio\core\io.py:43: **UserWarning: torchaudio._backend.set_audio_backend has been deprecated. With dispatcher enabled, this function is no-op.** You can remove the function call.
  torchaudio.set_audio_backend("soundfile")
C:\Users\valen\.conda\envs\diart\lib\site-packages\torch_audiomentations\utils\io.py:27: **UserWarning: torchaudio._backend.set_audio_backend has been deprecated. With dispatcher enabled, this function is no-op**. You can remove the function call.
  torchaudio.set_audio_backend("soundfile")
**The torchaudio backend is switched to 'soundfile'. Note that 'sox_io' is not supported on Windows.
The torchaudio backend is switched to 'soundfile'. Note that 'sox_io' is not supported on Windows.**
 
 There is any direction I can follow to solve such issue.
 Thanks

**Comment by juanmc2005** [comment]
Hi @vtontodonato, `diart.stream` expects a directory as `--output`. The name of the file will be the same as your audio file but with the `.rttm` extension.

Not sure why you aren't getting any errors though. I do get them on linux.
Let me know if this solves your problem

**Reply by vtontodonato** [reply]
... the problem mainly is that I'm a senior software developer, but quite new to python environment.
I managed to solve the first problem, it was libsound that was loaded wrong (32bit) from Anaconda, I find out how to reinstall the libsound. Now the system stop at the very first step. I'm trying to work in python directly.

pipeline = SpeakerDiarization()

gives an exception ... OSError: [WinError 6] invalid handle.
And of course the system stops working; very frustrating.

All the import statements works correctly.
from diart import SpeakerDiarization
from diart.sources import MicrophoneAudioSource
from diart.inference import StreamingInference
from diart.sinks import RTTMWriter

Thanks for any help.

**Reply by juanmc2005** [reply]
@vtontodonato can you debug the `__init__()` method of `SpeakerDiarization` line by line to see where it crashes exactly?

**Comment by vtontodonato** [comment]
.. not very sure on what is going on, but:
The __init__() method seems even not be called at all. Even if I put a breakpoint at the very first line of the method, the error occurs before it. 

Please take into account that I'm using visual studio code/python/anaconda for the very first time.

**Reply by juanmc2005** [reply]
The thing is, in order to understand what's causing this error, we need to pinpoint the exact line that's causing it. Maybe try debugging line by line from the very first import of your script
